extras/src/paych.rs

```diff
@@ -0,0 +1,183 @@
+    /// Max epoch beyond which the voucher cannot be redeemed
+    /// set to 0 means no timeout
+    pub time_lock_max: ChainEpoch,
+    /// (optional) Used by `to` to validate
+    #[serde(with = "serde_bytes")]
+    pub secret_pre_image: Vec<u8>,
+    /// (optional) Specified by `from` to add a verification method to the voucher
+    pub extra: Option<ModVerifyParams>,
+    /// Specifies which lane the Voucher merges into (will be created if does not exist)
+    pub lane: u64,
+    /// Set by `from` to prevent redemption of stale vouchers on a lane
+    pub nonce: u64,
+    /// Amount voucher can be redeemed for
+    #[serde(with = "bigint_ser")]
+    pub amount: BigInt,
+    /// (optional) Can extend channel min_settle_height if needed
+    pub min_settle_height: ChainEpoch,
+
+    /// (optional) Set of lanes to be merged into `lane`
+    pub merges: Vec<Merge>,
+
+    /// Sender's signature over the voucher (sign on none)
+    pub signature: Option<Signature>,
+}
+
+impl SignedVoucher {
+    pub fn signing_bytes(&self) -> Result<Vec<u8>, Error> {
+        /// Helper struct to avoid cloning for serializing structure.
+        #[derive(Serialize_tuple)]
+        struct SignedVoucherSer<'a> {
+            pub channel_addr: &'a Address,
+            pub time_lock_min: ChainEpoch,
+            pub time_lock_max: ChainEpoch,
+            #[serde(with = "serde_bytes")]
+            pub secret_pre_image: &'a [u8],
+            pub extra: &'a Option<ModVerifyParams>,
+            pub lane: u64,
+            pub nonce: u64,
+            #[serde(with = "bigint_ser")]
+            pub amount: &'a BigInt,
+            pub min_settle_height: ChainEpoch,
+            pub merges: &'a [Merge],
+            pub signature: (),
+        }
+        let osv = SignedVoucherSer {
+            channel_addr: &self.channel_addr,
+            time_lock_min: self.time_lock_min,
+            time_lock_max: self.time_lock_max,
+            secret_pre_image: &self.secret_pre_image,
+            extra: &self.extra,
+            lane: self.lane,
+            nonce: self.nonce,
+            amount: &self.amount,
+            min_settle_height: self.min_settle_height,
+            merges: &self.merges,
+            signature: (),
+        };
+        // Cbor serialize struct
+        to_vec(&osv)
+    }
+}
+
+/// Modular Verification method
+#[derive(Debug, PartialEq, Serialize_tuple, Deserialize_tuple)]
+pub struct ModVerifyParams {
+    pub actor: Address,
+    pub method: MethodNum,
+    pub data: Serialized,
+}
+
+/// Payment Verification parameters
+#[derive(Serialize_tuple, Deserialize_tuple)]
+pub struct PaymentVerifyParams {
+    pub extra: Serialized,
+    // TODO revisit these to see if they should be arrays or optional
+    #[serde(with = "serde_bytes")]
+    pub proof: Vec<u8>,
+}
+
+/// State channel parameters
+// REVIEW: do we still need proof ? (https://github.com/filecoin-project/specs-actors/blob/master/actors/builtin/paych/paych_actor.go#L80-L83)
```

Comment:
@rllola is this still relevant? Should we make an issue with a longer explanation for this?